### PR TITLE
ISPN-5647 ClassCastException on transactional cache

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-5647

### CODE REVIEW

File: core/src/test/java/org/infinispan/tx/GetRemoteEntryTxTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      ConfigurationBuilder builder = getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC, true);
+      builder.clustering().hash().numOwners(1);
+      createCluster(builder, 2);
+      waitForClusterToForm();
+   }
+
+   public void testGetRemoteEntry() throws Exception {
```
Comment (wburns) [inline core/src/test/java/org/infinispan/tx/GetRemoteEntryTxTest.java]: Lets add a  local one too just for good measure :)

**Comment by gustavonalle** [comment]
PR Updated

File: core/src/test/java/org/infinispan/tx/GetEntryTxTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   public void testGetEntry() throws Exception {
+      Cache<MagicKey, String> cache = cache(0);
+
+      MagicKey localKey = new MagicKey(cache(0));
+      MagicKey remoteKey = new MagicKey(cache(1));
+
+      cache.put(localKey, localKey.toString());
+      cache.put(remoteKey, remoteKey.toString());
+
+      assertNotNull(cache.getAdvancedCache().getCacheEntry(localKey));
```
Comment (wburns) [inline core/src/test/java/org/infinispan/tx/GetEntryTxTest.java]: I would normally prefer them to be separate tests, but that is okay.

**Comment by wburns** [comment]
Pulling...

**Comment by wburns** [comment]
Integrated into master, thanks @gustavonalle !

**Comment by wburns** [comment]
Testing 7.2.x real quick then pushing there too.

**Pull request closed without merging**